### PR TITLE
reflect without missile!

### PR DESCRIPTION
**Pull request by qndel**
This gets rid of all the bugs and even allows stacking reflect charges. Inspired by berserk.
Tested in single and multi, but since it's a completely new approach, I'd appreciate some testing from you too : )
Look how much code got trashed 😀 
If this gets merged, we can think about rewriting manashield same way, as iirc it stored some variables in the missile but do we really need them after we've included manashield in the damage calculation itself?

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -3221,7 +3225,7 @@ void RemovePlrMissiles(int pnum)
 		if (missile._mitype == MIS_STONE && missile._misource == pnum) {
 			Monsters[missile._miVar2]._mmode = (MON_MODE)missile._miVar1;
 		}
-		if (IsAnyOf(missile._mitype, MIS_MANASHIELD, MIS_REFLECT) && missile._misource == pnum) {
+		if (IsAnyOf(missile._mitype, MIS_MANASHIELD) && missile._misource == pnum) {
```
Comment (glebm) [inline Source/player.cpp:3228]:
```suggestion
		if (missile._mitype == MIS_MANASHIELD && missile._misource == pnum) {
```

Comment (qndel) [reply Source/player.cpp:3228]: I knew about this but left it for future expansion ;p

**Comment by Chance4us** [comment]
I've already tested a little and haven't had any problems so far.

**Comment by qndel** [comment]
> I've already tested a little and haven't had any problems so far.

Can you join discord? :p It's much faster to chat than on github :D

**Review by AJenbo — APPROVED** [review]

**Comment by Chance4us** [comment]
> > I've already tested a little and haven't had any problems so far.
> 
> Can you join discord? :p It's much faster to chat than on github :D

Yes great, i will notify you on discord.

**Pull request merged**